### PR TITLE
feat: observability in snyk code [NEBULA-399]

### PR DESCRIPTION
**Pull request by metju90**
https://snyksec.atlassian.net/browse/NEBULA-339

Ignore the request ID difference in one of the screenshots 🙃 

image
image
image

### CODE REVIEW

File: src/lib/plugins/sast/index.ts [hunk 1]
```diff
@@ -12,8 +13,7 @@ import { FailedToRunTestError, NoSupportedSastFiles } from '../../errors';
 import { jsonStringifyLargeObject } from '../../json';
 import * as analytics from '../../analytics';
 
-debugLib.enable('snyk-code');
-const debug = debugLib('snyk-code');
```
Comment (pkey) [inline src/lib/plugins/sast/index.ts:16]: question: why change this? not sure what it does though.

Comment (metju90) [reply src/lib/plugins/sast/index.ts:16]: The removed code fixed an issue with not printing debugging code from [here](https://github.com/snyk/snyk/pull/2748/files#diff-2284575013fd6588d30774b796013333e67e588d92cf0b4fee2508d926419614L84). but it fixed it too good, so good that it was outputting printing debug info even without the `-d` flag.

Now it's working as intended, debug info only when `-d` is passed.

Comment (pkey) [reply src/lib/plugins/sast/index.ts:16]: gotcha

Comment (metju90) [reply src/lib/plugins/sast/index.ts:16]: TBH I didn't fully understand how our debug library works, but now we have the desired behaviour ✌️

**Review by pkey — APPROVED** [review]

**Comment by pkey** [comment]
@metju90 does [this](https://github.com/snyk/snyk/pull/2756) fix the issue you were seeing [here](https://github.com/snyk/snyk/pull/2748/files#diff-2284575013fd6588d30774b796013333e67e588d92cf0b4fee2508d926419614L15)?

**Comment by metju90** [comment]
> @metju90 does [this](https://github.com/snyk/snyk/pull/2756) fix the issue you were seeing [here](https://github.com/snyk/snyk/pull/2748/files#diff-2284575013fd6588d30774b796013333e67e588d92cf0b4fee2508d926419614L15)?

Good catch, I will check it out 👮

**Comment by metju90** [comment]
@pkey so apparently with me using `debug(snyk-code)` and doing `debug.enable` I broke their stuff. 

I've asked the product [here](https://snyk.slack.com/archives/C01KLME457X/p1644929792211759) on whether we should use `snyk-code` or `snyk` to debug when using Snyk Code.

**Review by Avishagp — APPROVED** [review]

**Pull request merged**